### PR TITLE
release-21.1: timeutil: better error reporting

### PR DESCRIPTION
**Pull request by knz**
Backport 4/4 commits from #63327.

/cc @cockroachdb/release

---

### CODE REVIEW

**Review by rafiss — APPROVED** [review]
> Reviewed 2 of 2 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/63540)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @rafiss)

**Comment by knz** [comment]
I'll hold off until 21.1.0 is picked before merging this

**Pull request merged**
